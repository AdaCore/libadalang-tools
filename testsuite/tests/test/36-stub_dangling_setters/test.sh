#! /bin/bash

# Generate a stubbing harness with the old sources, containing a custom type
gnattest -P prj.gpr -q -XSRC_DIR=old --stub

# Re-generate the harness but with the custom type dissapearing. Any remaining
# references to it will cause a compilation failure
gnattest -P prj.gpr -XSRC_DIR=new --stub

# Build the harness to check that the unused setters are not causing any
# problems.
gprbuild -P obj/gnattest_stub/harness/test_drivers.gpr -q -XSRC_DIR=new
