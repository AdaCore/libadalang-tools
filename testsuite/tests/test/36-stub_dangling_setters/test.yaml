description:
  Check that dangling setters in a stubbing harness do not cause any compilation
  issues. They used to not be commented out, which could result in lingering
  references to types or objects no longer present in the codebase.

driver: gnattest_driver
