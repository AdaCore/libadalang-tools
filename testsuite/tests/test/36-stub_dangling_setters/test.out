Units remaining: 4     Units remaining: 3      warning: (gnattest) Dep.Test_Data.Tests has dangling test(s)
Units remaining: 2     Units remaining: 1      warning: (gnattest) Dep has dangling element(s)
 warning: (gnattest) Dep.Stub_Data has dangling setter spec(s)
 warning: (gnattest) Dep.Stub_Data has dangling setter body(ies)

